File: C++Projects/6276_cppclase1-2_archivosdistintos/Perro.cpp
#include "Perro.h"

// Constructor
Perro::Perro(string nombre) : Mamifero(nombre) {}

// Métodos provenientes de Animal
void Perro::expresarse() {
    cout << "Guau" << endl;
}
string Perro::recorridoEnMetrosPorSegundo() {
    return "Recorro dos metros en un segundo";
}
void Perro::alimentarse(string tipo){
    if(tipo == "hogareño"){
        cout << "Vivo en un hogar, me alimento de concentrado" << endl;
    }
    else if(tipo == "callejero"){
        cout << "Me alimento de lo que encuentro en la calle" << endl;
    }
    else {
        cout << "No se ha definido de qué me alimento" << endl;
    }
}

// Getter y Setter
string Perro::getEdad(){
    return to_string(edad) + " years";
}
void Perro::setEdad(int edad){
    this->edad = edad;
}

File: JavaProjects/Clase6276-3-HerenciaMultiple/src/Animales/Perro.java
package Animales;

import ParentClasses.Animal;
import ParentClasses.Mamifero;

public class Perro extends Mamifero implements Animal {
    public Perro(String nombre){
        super(nombre);
    }

    @Override
    public void expresarse() {
        System.out.println("Guau");
    }

    @Override
    public String recorridoEnMetrosPorSegundo(){
        return "¡Corro 2 metros en un segundo!";
    }

    @Override
    public void alimentarse(String tipo){
        if (tipo.equals("hogareño")){
            System.out.println("Vivo en un hogar, me alimento de concentrado");
        }
        else if (tipo.equals("callejero")){
            System.out.println("Me alimento de lo que encuentro en la calla");
        }
        else {
            System.out.println("No me han definido de qué me alimento");
        }
    }
}
